SpatialGDK/Source/SpatialGDKServices/Private/SpatialPackageManager.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialPackageManager.h"
+#include "SpatialGDKServicesConstants.h"
+#include "Async/Async.h"
+#include "Misc\MonitoredProcess.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialPackageManager);
+
+#define LOCTEXT_NAMESPACE "FSpatialPackageManager"
+
+FSpatialPackageManager::FSpatialPackageManager() {}
+
+void FSpatialPackageManager::Init() {}
+
+void FSpatialPackageManager::TryFetchRuntimeBinary(FString RuntimeVersion)
+{
+
+	UE_LOG(LogSpatialPackageManager, Log, TEXT("Trying to fetch runtime version %s"), *RuntimeVersion);
+
+	FString RuntimePath = FString::Printf(TEXT("%s/runtime/version-%s/%s"), *SpatialGDKServicesConstants::GDKProgramPath, *RuntimeVersion,
+										  *SpatialGDKServicesConstants::RuntimeExe);
+
+	// Check if the binary already exists for a given version
+	if (FPaths::FileExists(RuntimePath))
+	{
+		UE_LOG(LogSpatialPackageManager, Log, TEXT("Runtime Binaries already exist."));
+		return;
+	}
+
+	// If it does not exist then fetch the binary using `spatial worker package retrieve`
+	// Download the zip to // UnrealGDK\SpatialGDK\Binaries\ThirdParty\Improbable\Programs\Runtime\*version* and unzip
+	FString Params =
+		FString::Printf(TEXT("package retrieve runtime x86_64-win32 %s runtime/version-%s --unzip"), *RuntimeVersion, *RuntimeVersion);
```

Comment:
Platform version should probably be a constant. Also, I think we should make the platform version a `GetPlatformVersion` function in `SpatialGDKServicesConstants`, then we can make it platform agnostic in the future

Reply:
x86_64-win32 - is this considered the Platform version?

Reply:
added!